src/app/components/file-uploader/file-uploader.component.ts

```diff
@@ -0,0 +1,157 @@
+import { Component, Input, ViewChild, ElementRef, OnInit } from '@angular/core';
+import { UploadFile } from 'src/app/models/domain/uploadFile';
+import { FileUploaderService } from 'src/app/services/file-uploader.service';
+import { map } from 'rxjs/operators';
+import { HttpEventType } from '@angular/common/http';
+import { forkJoin, Observable, of } from 'rxjs';
+import { AlertConfig } from '../../models/internal/alert-config';
+import { AlertType } from '../../models/internal/alert-type';
+import { AlertService } from '../../services/alert.service';
+
+@Component({
+  selector: 'app-file-uploader',
+  templateUrl: './file-uploader.component.html',
+  styleUrls: ['./file-uploader.component.scss']
+})
+export class FileUploaderComponent {
+
+  @Input() acceptMultiple: boolean;
+  @Input() acceptedTypes: Array<String>;
+  @ViewChild('fileDropRef') fileInput: ElementRef;
+
+  maxFileSize = 5242880;
+
+  constructor(private uploadService: FileUploaderService,
+              private alertService: AlertService) { }
+
+  files: Array<UploadFile> = new Array<UploadFile>();
+
+  /**
+   * on file drop handler
+   */
+  onFileDropped(event) {
+    this.prepareFilesList(event);
+  }
+
+  /**
+   * handle file from browsing
+   */
+  fileBrowseHandler(files) {
+     if (files.files !== this.files) {
+      this.prepareFilesList(files.files);
+     }
+  }
+
+  /**
+   * Delete file from files list
+   * @param index (File index)
+   */
+  deleteFile(index: number) {
+    this.files.splice(index, 1);
+    this.fileInput.nativeElement.value = '';
+  }
+
+  private prepareFilesList(files: Array<UploadFile>) {
+    // If the user can only select 1 image we want to reset the array
+    if ( !this.acceptMultiple) {
+      this.files = [];
+    }
+    for (const file of files) {
+      if (file.size < this.maxFileSize) {
+        if (this.acceptedTypes.includes(file.type)) {
+          this.generatePreview(file);
+          this.formatBytes(file);
+          this.files.push(file);
+        } else {
+          const alertConfig: AlertConfig = {
+            type: AlertType.danger,
+            preMessage: 'This file type is not allowed',
+            mainMessage: `File ${file.name} is of type that is not allowed please pick one of these ${this.acceptedTypes.join(', ')}`,
+            dismissible: true,
+            timeout: this.alertService.defaultTimeout
+          };
+          this.alertService.pushAlert(alertConfig);
+          this.deleteFile(this.files.indexOf(file));
+        }
+      } else {
+        const alertConfig: AlertConfig = {
+          type: AlertType.danger,
+          preMessage: 'This file is too big',
+          mainMessage: `File ${file.name} is too big, the max size is 5mb`,
+          dismissible: true,
+          timeout: this.alertService.defaultTimeout
```

Comment:
You can leave the timeout property unset if you are using the default timeout. It is a nullable property and the service will already use the default timeout if none is set.